spec/internal/ssl.lua

```diff
@@ -259,9 +259,7 @@ function SSL:send(s)
 end
 
 function SSL:close()
-  if C.SSL_shutdown(self.ctx) ~= 1 then
-    return nil, format_error("SSL_shutdown")
-  end
+  self.sock:close()
```

Comment:
To reviewers: `SSL_shutdown` only does a half-close, so the connection can still be used after being closed [here](https://github.com/Kong/kong/blob/9ae36382db4b9485f8889b8cd80d66a0885c86e7/spec/internal/server.lua#L101). This causes some tests in tcp-log_spec.lua and reports_spec.lua to fail because the connection established by a previous test case is reused by the next test case.

Reply:
LGTM, but one concern about why we need internal SSL implementation( not related this PR)

Reply:
I see